org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/hotspots/SecurityHotspotsHandlerServer.java

```diff
@@ -167,6 +174,10 @@ public void handle(ClassicHttpRequest request, ClassicHttpResponse response, Htt
     }
   }
 
+  private static boolean isTrustedServer(String serverOrigin) {
+    return SonarLintCorePlugin.getServersManager().getServers().stream().anyMatch(s -> s.getHost().contains(serverOrigin));
```

Comment:
I have a doubt about this `contains`. Could we imagine a scenario where an attacker could use an origin that has the registered server URL as a substring ?

Reply:
Thanks for the comment, I revisited using `startsWith` instead (which more closely matches the actual spec) and commented the code for future reference.